### ISSUE TITLE
Connect to testnet correctly, based on configuration

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/zebra#247]
Once #52 is complete and we actually perform version negotiation, we should allow configuring to either mainnet or testnet based on a config flag.  This is probably easy to do as part of that work.

**Comment by oxarbitrage** [comment]
I am doing some research to implement this and i have a problem. 

I know that changing the default config file option from `network = 'Mainnet'` to `network = 'Testnet'` works for the **connect** command. It should work also for **seed** and **start** however didn't tested yet. So this issue can be considered done when there is a command line option that do the same as what manually can be done with the config file.

As there are 3 subcommands that will make use of this flag, it will be nice to have it as a global option however i am unsure if this is currently possible. The global flags are hardcoded in abscissa at https://github.com/iqlusioninc/abscissa/blob/develop/core/src/command/entrypoint.rs#L15-L25 and unless there is something rusty i am missing i dont see how we can overwrite that from zebra side.

A doable option that i can think on is to add a `--network` option inside each subcommand that will connect to somewhere(connect, seed, start) however not sure if that will be acceptable.

Please let me know what do you think, thanks :)

**Comment by yaahc** [comment]
I think it might be possible to add top level arguments if we change https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/commands.rs#L25 to a diff name and replace it with a struct with the current name then take the old enum and make it the command option of the new struct that has the top level flag like this:

```rust
// Define options for the program.
#[derive(Debug, Options)]
struct MyOptions {
    // Options here can be accepted with any command (or none at all),
    // but they must come before the command name.
    #[options(help = "print help message")]
    help: bool,
    #[options(help = "be verbose")]
    verbose: bool,

    // The `command` option will delegate option parsing to the command type,
    // starting at the first free argument.
    #[options(command)]
    command: Option<Command>,
}
```

**Comment by hdevalence** [comment]
I'm not sure if we should have a command-line flag to select which network to connect to rather than using the flag in the config file.  One reason is that there may be other settings that should change alongside the network (like the listening port, which is usually different on testnet vs. mainnet).

@oxarbitrage did you test using the `connect` command to connect to a testnet-running `zcashd` instance?  I think the motivation for the ticket was that when we implemented the network stack we tested against mainnet, so there might be hidden problems with connecting to the testnet.

In terms of `connect`, `seed`, and `start`, the `connect` command is just stub code, and we're planning to rip it out before we do a release.  The `start` command will be the main priority, but right now, it doesn't really work, because we're still putting the pieces together.  However, almost all the networking logic is handled internally to `zebra-network`, so testing the stub code in `connect` should reveal most of the problems if there are any.

**Comment by teor2345** [comment]
There might also be some changes we need to make in zebra-consensus, like using a different checkpoint list. (And maybe checking mainnet vs testnet in validation?)

But none of that is implemented yet.

**Comment by oxarbitrage** [comment]
Thank you @hdevalence There are probably more tests needed, initially when i wrote above i just tried connecting to mainnet and testnet by changing the configuration file. When this is done the program correctly connect to the corresponding peers so i assumed this was working and interpreted the problem after that wrongly, apologies.

For that simple test done i have some logs, here is the testnet:

https://github.com/oxarbitrage/logs/blob/master/zcash/zebra_connect_testnet.txt

And here the mainnet. Note the difference at line 6:

https://github.com/oxarbitrage/logs/blob/master/zcash/zebra_connect_mainnet.txt

A command line flag might not be a good idea but I do think that we will need to add more configure global flags at some point so maybe that discussion can be moved as something more general to a separated issue.

**Comment by hdevalence** [comment]
Sounds good, looks like we can connect to the testnet, then!  Doing the testing & changes that @teor2345 mentioned can't happen yet because we haven't implemented our consensus logic, but at least not having any problems at the networking layer is a good start, so I'm going to close this issue as resolved.
